src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitWorkerService.java

```diff
@@ -0,0 +1,194 @@
+public class VoltageInitWorkerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VoltageInitWorkerService.class);
+
+    private NetworkStoreService networkStoreService;
+
+    private VoltageInitResultRepository resultRepository;
+
+    private Map<UUID, CompletableFuture<OpenReacResult>> futures = new ConcurrentHashMap<>();
+
+    private Map<UUID, VoltageInitCancelContext> cancelComputationRequests = new ConcurrentHashMap<>();
+
+    private Set<UUID> runRequests = Sets.newConcurrentHashSet();
+
+    private final Lock lockRunAndCancelVoltageInit = new ReentrantLock();
+
+    private final Executor threadPool = ForkJoinPool.commonPool();
+
+    @Autowired
+    NotificationService notificationService;
+
+    public VoltageInitWorkerService(NetworkStoreService networkStoreService, VoltageInitResultRepository resultRepository) {
+        this.networkStoreService = Objects.requireNonNull(networkStoreService);
+        this.resultRepository = Objects.requireNonNull(resultRepository);
+    }
+
+    private Network getNetwork(UUID networkUuid, String variantId) {
+        Network network;
+        try {
+            network = networkStoreService.getNetwork(networkUuid, PreloadingStrategy.ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW);
+            String variant = StringUtils.isBlank(variantId) ? VariantManagerConstants.INITIAL_VARIANT_ID : variantId;
+            network.getVariantManager().setWorkingVariant(variant);
+        } catch (PowsyblException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage());
+        }
+        return network;
+    }
+
+    private Network getNetwork(UUID networkUuid, List<UUID> otherNetworkUuids, String variantId) {
+        Network network = getNetwork(networkUuid, variantId);
+        if (otherNetworkUuids.isEmpty()) {
+            return network;
+        } else {
+            List<Network> otherNetworks = otherNetworkUuids.stream().map(uuid -> getNetwork(uuid, variantId)).collect(Collectors.toList());
+            List<Network> networks = new ArrayList<>();
+            networks.add(network);
+            networks.addAll(otherNetworks);
+            MergingView mergingView = MergingView.create("merge", "iidm");
+            mergingView.merge(networks.toArray(new Network[0]));
+            return mergingView;
+        }
+    }
+
+    private OpenReacResult run(VoltageInitRunContext context, UUID resultUuid) throws ExecutionException, InterruptedException {
+        Objects.requireNonNull(context);
+
+        LOGGER.info("Run voltage init...");
+        Network network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids(), context.getVariantId());
+
+        CompletableFuture<OpenReacResult> future = runVoltageInitAsync(context, network, resultUuid);
+
+        return future == null ? null : future.get();
+    }
+
+    public CompletableFuture<OpenReacResult> runVoltageInitAsync(VoltageInitRunContext context, Network network, UUID resultUuid) {
```

Comment:
context is not used

Reply:
Yes, but i kept it as it gonna be used in next developments. The parameters will be gotten in the context. Ok to keep it ?

Reply:
yes!